astrocut/fits_cutout.py

```diff
@@ -439,6 +429,39 @@ def write_as_fits(self, output_dir: Union[str, Path] = '.', cutout_prefix: str =
 
             log.debug('Cutout filepaths: {}'.format(cutout_paths))
             return cutout_paths
+
+    def write_as_zip(self, output_dir: Union[str, Path] = '.', filename: Union[str, Path, None] = None) -> str:
+        """
+        Package the FITS cutouts into a zip archive without writing intermediate files.
+
+        Parameters
+        ----------
+        output_dir : str | Path, optional
+            Directory where the zip will be created. Default '.'.
+        filename : str | Path | None, optional
+            Name (or path) of the output zip file. If not provided, defaults to
+            'cutouts_{YYYYmmdd_HHMMSS}.zip'. If provided without a '.zip' suffix,
+            the suffix is added automatically.
+
+        Returns
+        -------
+        str
+            Path to the created zip file.
+        """
+        def build_entries():
+            if self._single_outfile:
+                # Mirror the single-file naming used by write_as_fits
+                arcname = self._make_cutout_filename('cutout')
+                hdu = self.fits_cutouts[0]
+                yield arcname, hdu
+            else:
+                # One file per input; mirror write_as_fits naming
+                for i, file in enumerate(self.hdu_cutouts_by_file):
+                    arcname = self._make_cutout_filename(Path(file).stem)
+                    hdu = self.fits_cutouts[i]
+                    yield arcname, hdu
```

Comment:
Is there a reason these functions are defined within the `write_as_zip` method on each class, as opposed to an (overridden) method on the class itself?  Besides the one for asdf cutouts, they seem fairly separable.  Not suggesting a change necessarily, just curious.